src/elementPlugins/shortText/src/ApplicationView.tsx

```diff
@@ -9,43 +9,70 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
   isEditable,
   isRequired,
   allResponses,
+  evaluator,
 }) => {
-  const [validationMessage, setValidationMessage] = useState('')
+  const [validationMessageDisplay, setValidationMessageDisplay] = useState('')
   const [value, setValue] = useState(initialValue?.text)
+  const [isValid, setIsValid] = useState<boolean>()
+  const [responses, setResponses] = useState({
+    thisResponse: undefined,
```

Comment:
`thisResponse: initialValue?.text` ?

Hmm, i am wondering if this is a tad too much boilerplate for a plugin. I have a couple of ideas. Should be fine for now thought

Reply:
I'm open to ideas, we just need to make sure we meet the following two criteria:
- we need the evaluator to have access to an object that will change on every keystroke (the main `allResponses` object only gets updated onLoseFocus)
- evaluator expressions shouldn't have to know the code of their own question when defining queries (i.e. you should be able to copy and paste an evaluator validation query from one question to another without having to change anything)

Maybe we can abstract more of this logic away to the ApplicationViewWrapper and keep the plugins themselves quite limited?